From 3c886ddf43b192c3548e573cbfda89505be9371c Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Tue, 4 Jul 2017 09:30:49 +1000
Subject: [PATCH 07/10] context: Enable atomic updates for PGO generation

New flag in GCC 7 -fprofile-update=atomic prevents creation of corrupted
profiles created during instrumentation, but with a speed penalty.

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 ypkg2/ypkgcontext.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ypkg2/ypkgcontext.py b/ypkg2/ypkgcontext.py
index b6b4396..3da5b50 100644
--- a/ypkg2/ypkgcontext.py
+++ b/ypkg2/ypkgcontext.py
@@ -32,7 +32,8 @@ BIND_NOW_FLAGS = ["-Wl,-z,now", "-Wl,-z -Wl,relro", "-Wl,-z -Wl,now"]
 SIZE_FLAGS = "-Os -ffunction-sections"
 
 # GCC PGO flags
-PGO_GEN_FLAGS = "-fprofile-generate -fprofile-dir=\"{}\""
+PGO_GEN_FLAGS = "-fprofile-generate -fprofile-dir=\"{}\" " \
+                "-fprofile-update=atomic"
 PGO_USE_FLAGS = "-fprofile-use -fprofile-dir=\"{}\" -fprofile-correction"
 
 # Clang can handle parameters to the args unlike GCC
-- 
2.13.3

